pkg/projectfile/projectfile.go

```diff
@@ -824,7 +824,9 @@ func GetProjectFilePath() (string, *failures.Failure) {
 	for _, getProjectFilePath := range lookup {
 		path, fail := getProjectFilePath()
 		if fail != nil {
-			return "", fail
+			// log error and try the next function
+			logging.Error("Failed to get project file: %v", fail.ToError())
+			continue
```

Comment:
The contract here is that a failure should be raised. If the issue is considered benign then the `getProjectFile..` function would not be returning an error at all. This change is adding complexity to solve a problem that's unrelated to this function.

I think the current behaviour is proper; if your environment is corrupted it should be raised to the user so they can fix it. The problem we're fixing is that this might happen during installation. Maybe for the `state export` call we could ensure in the installer it's called with a clean environment and from a temp path that won't have a project file?

Reply:
That sounds good.